### PR TITLE
feat(npu): add Ascend 910B support

### PR DESCRIPTION
**Pull request by SolenoidWGT**
## Motivation

Support Ascend 910B training, remove `torch.cuda` related code and introduce `accelerator.py` to manage different accelerator backends.

## Modification

TODO

## BC-breaking (Optional)

TODO

## Use cases (Optional)

TODO

## Checklist

**Before PR**:

- [ ] Pre-commit or other linting tools are used to fix the potential lint issues.
- [ ] Bug fixes are fully covered by unit tests, the case that causes the bug should be added in the unit tests.
- [ ] The modification is covered by complete unit tests. If not, please add more unit test to ensure the correctness.
- [ ] The documentation has been modified accordingly, like docstring or example tutorials.

**After PR**:

- [ ] If the modification has potential influence on downstream or other related projects, this PR should be tested with those projects.
- [ ] CLA has been signed and all committers have signed the CLA in this PR.

### CODE REVIEW

**Comment by sunpengsdu** [comment]
- fix CI
- maybe need to set a default backend
- in some cases, use_flash_attn is bounded to GPU. For example, in the pipeline scheduler, when we set use_flash_attn, it actually required to be cuda backend and use packed data. So in these cases, maybe we need to set backend=="GPU" and use_flash_attn=true -> pack dataset

File: internlm/train/pipeline.py [hunk 1]
```diff
@@ -291,10 +291,11 @@ def initialize_optimizer(model: Union[nn.Module, nn.ModuleList], isp_communicato
...
     adam_extra_kwargs = {}
     # set fused=True to avoid nan grad norm when model size is larger and use_fp32_norm=True
-    if torch.__version__ >= "2.1.0":
-        adam_extra_kwargs["fused"] = True
+    # if torch.__version__ >= "2.1.0":
+    #     adam_extra_kwargs["fused"] = True
+    import torch_npu
 
-    naive_optimizer = torch.optim.AdamW(
+    naive_optimizer = torch_npu.optim.NpuFusedAdamW(
```
Comment (huangting4201) [inline internlm/train/pipeline.py]: 这里要跟GPU情况下兼容吧，做个if else判断

Comment (SolenoidWGT) [reply internlm/train/pipeline.py]: Done

File: internlm/accelerator/cuda_accelerator.py [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+
+    def default_generator(self, device_index):
+        return torch.cuda.default_generators[device_index]
+
+    # Streams/Events
+    @property
+    def Stream(self):
+        return torch.cuda.Stream
+
+    def stream(self, stream):
```
Comment (sunpengsdu) [inline internlm/accelerator/cuda_accelerator.py]: 函数命 same with 参数 stream

Comment (SolenoidWGT) [reply internlm/accelerator/cuda_accelerator.py]: Done

**Review by sunpengsdu — APPROVED** [review]
